pkg/fanal/analyzer/pkg/apk/apk.go

```diff
@@ -63,6 +64,7 @@ func (a alpinePkgAnalyzer) parseApkInfo(scanner *bufio.Scanner) ([]types.Package
 		// check package if paragraph end
 		if len(line) < 2 {
 			if !pkg.Empty() {
+				pkg.Identifier = purl.NewPackageIdentifier(types.TargetType(analyzer.TypeApk), nil, pkg)
```

Comment:
We do this when merge layers.
It doesn't make sense.

Reply:
same for `dpkg` and `rpm`

Reply:
done at https://github.com/aquasecurity/trivy/pull/5439/commits/19fbf7b6f7d528e44d1cf720f78727c2956b4e27